### PR TITLE
Segregate graphics and non-graphics tests

### LINKED ISSUE #378: Segregate graphics and non-graphics tests
Currently, graphics tests produce random memory access errors, crashing test suites. 
A recent example would be #377 https://github.com/extendr/extendr/runs/5206578831?check_suite_focus=true#step:11:121
Crashes are observed on both Windows and Linux machines, but because the number of Linux instances we spin is larger than any other systems, the majority of recent failures originate from Linux runners.

I proposed (somewhere, can't find the link) that we separate graphics tests from all other tests, i.e. remove it from `tests-all` feature.
This will allow us to run regular tests in parallel (as we do now) and then run only graphics tests with `--test-threads 1`. This will probably help us to diagnose graphics problems more easily.

TODO:
- [x] Finish #373
- [ ] Ensure `graphics` is not part of the `tests-all` set and is not tested 'by default' so this CI step does not test graphics https://github.com/extendr/extendr/blob/678f02c51573563be60969691612a9c4720123aa/.github/workflows/test.yml#L157
- [ ] Modify CI to execute additional `cargo test --features graphics -- --test-threads 1` 
- [ ] Remove `--test-threads 1` from remaining tests https://github.com/extendr/extendr/blob/678f02c51573563be60969691612a9c4720123aa/.github/workflows/test.yml#L417-L421

I am mentioning and assigning it to @yutannihilation. I consider this issue to be a priority as it affects and complicates the process of code review.

### PR DESCRIPTION
**Pull request by yutannihilation**
Close #378 

I created `tests` feature, which is `tests-all` minus `graphics`, and use it on the CI.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -414,8 +415,9 @@ jobs:
 
             ci-cargo +$toolchain test --manifest-path extendr-engine/Cargo.toml --features tests-all $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture -ActionName "Test extendr-engine \w tests-all for $target target"
 
-            # TODO: --test-threads=1 is needed because graphics-related tests need to run in serial
-            ci-cargo +$toolchain test --manifest-path extendr-api/Cargo.toml --features tests-all $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture --test-threads=1 -ActionName "Test extendr-api \w tests-all for $target target"
+            ci-cargo +$toolchain test --manifest-path extendr-api/Cargo.toml --features tests $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture -ActionName "Test extendr-api \w tests-all for $target target"
```
Comment (Ilia-Kosenkov) [inline .github/workflows/test.yml:418]:
```suggestion
            ci-cargo +$toolchain test --manifest-path extendr-api/Cargo.toml --features tests $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture -ActionName "Test extendr-api \w `tests` for $target target"
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -414,8 +415,9 @@ jobs:
...
+            # graphics test requires --test-threads=1
+            ci-cargo +$toolchain test --manifest-path extendr-api/Cargo.toml --features graphics $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture --test-threads=1 -ActionName "Test extendr-api \w tests-all for $target target"
```
Comment (Ilia-Kosenkov) [inline .github/workflows/test.yml:420]:
```suggestion
            ci-cargo +$toolchain test --manifest-path extendr-api/Cargo.toml --features graphics $(if($target -ne 'default') {"--target=$target"} ) $env:EXTRA_ARGS '--' --nocapture --test-threads=1 -ActionName "Test extendr-api \w `graphics` for $target target"
```

**Comment by Ilia-Kosenkov** [comment]
We definitely need to document various `test` features for developers (us), in addition to documenting user-facing feature gates.
Maybe as a separate PR.

File: .github/workflows/test.yml (hunk 1 above)
Comment (yutannihilation) [reply .github/workflows/test.yml:418]: Aw, thanks for catching!

**Review by Ilia-Kosenkov — APPROVED** [review]
I like when all CI checks are ✔ :) Seems to work as expected.

**Comment by yutannihilation** [comment]
Thanks!

> We definitely need to document various test features for developers (us), in addition to documenting user-facing feature gates.

I'll try this, but probably it won't be very soon, as I'm not sure where to document it...

**Pull request merged**